runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -98,6 +99,44 @@ public DataflowPipelineJob run(Pipeline pipeline) {
     return run(pipeline, runner);
   }
 
+  /**
+   * Enforces that a cancel() call on the job is done at most once - as a workaround for
+   * Dataflow service's current bugs with multiple cancellation, where it may sometimes
+   * return an error when cancelling a job that was already cancelled, but still report
```

Comment:
It does beg the question - why not make this the cancel behavior of `DataflowPipelineJob`?